tests/src/Kernel/BclIconPathOverrideTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\Kernel;
+
+use Drupal\Core\Theme\ThemeManagerInterface;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Tests the bcl_icon_path theme override functionality.
+ */
+class BclIconPathOverrideTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'system',
+  ];
+
+  /**
+   * The theme manager service.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected ThemeManagerInterface $themeManager;
+
+  /**
+   * Set up the test environment.
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Initialize the theme manager service.
+    $this->themeManager = $this->container->get('theme.manager');
+
+    // Install the theme. Subsequently, this installs also the base themes.
+    $this->container->get('theme_installer')->install(['oe_bootstrap_theme', 'oe_bootstrap_theme_test_subtheme']);
+    $this->config('system.theme')->set('default', 'oe_bootstrap_theme_test_subtheme')->save();
+  }
+
+  /**
+   * Tests that the bcl_icon_path is correctly overridden in the subtheme.
+   */
+  public function testBclIconPathOverride(): void {
+    // Retrieve the active theme.
+    $theme = $this->themeManager->getActiveTheme();
+
+    // Access the bcl_icon_path from the theme info file.
+    $themeInfo = $theme->getExtension()->info;
+    $bclIconPath = $themeInfo['openeuropa']['bootstrap_component_library']['bcl_icon_path'];
+
+    // Check if the paths match.
+    $this->assertEquals('assets/icons/bcl-default-icons-test.svg', $bclIconPath);
+
+    // Create a renderable array to simulate the theme preprocess.
+    $variables = [];
+
+    // Call the preprocess function manually.
+    oe_bootstrap_theme_preprocess($variables);
+
+    // Extract the path set in the preprocess function.
+    $actualIconPath = $variables['bcl_icon_path'];
+    $expectedIconPath = base_path() . $theme->getPath() . '/' . $bclIconPath;
+
+    // Check if the paths match.
+    $this->assertEquals($expectedIconPath, $actualIconPath);
+  }
+
+  /**
+   * Tests the fallback when the bcl_icon_path is not set in the subtheme.
+   */
+  public function testBclIconPathFallback(): void {
+    $this->container->get('theme_installer')->install(['oe_bootstrap_theme_test_subtheme_extra']);
+    $this->config('system.theme')->set('default', 'oe_bootstrap_theme_test_subtheme_extra')->save();
+
+    // Retrieve the active theme.
+    $theme = $this->themeManager->getActiveTheme();
+
+    // Access the bcl_icon_path from the theme info file.
+    $themeInfo = $theme->getExtension()->info;
+
+    // Optionally, check if the variable is empty if it exists.
+    $bclIconPath = $themeInfo['openeuropa']['bootstrap_component_library']['bcl_icon_path'] ?? NULL;
+    $this->assertNull($bclIconPath);
```

Comment:
We don't need to run these checks, we don't need to test our test data. We only need to run the preprocess and compare it with a static value that we know it's in the theme info file, like \Drupal\Tests\oe_bootstrap_theme_helper\Kernel\BclComponentsOverrideTest::bclComponentsOverridingTestCasesProvider.

Reply:
refactored it